src/deepsparse/server/openai_server.py

```diff
@@ -232,7 +232,28 @@ async def create_completion(raw_request: Request):
 
             request_id = f"cmpl-{random_uuid()}"
             created_time = int(time.time())
-            prompt = request.prompt
+
+            if isinstance(request.prompt, list):
+                if len(request.prompt) == 0:
+                    return create_error_response(
+                        HTTPStatus.BAD_REQUEST, "please provide at least one prompt"
+                    )
+                first_element = request.prompt[0]
+                if isinstance(first_element, int):
+                    # There is just a single tokenized prompt
+                    prompt = pipeline.tokenizer.decode(request.prompt)
+                elif isinstance(first_element, (str, list)):
+                    if len(request.prompt) > 1:
+                        return create_error_response(
+                            HTTPStatus.BAD_REQUEST,
+                            "multiple prompts in a batch is not currently supported",
+                        )
+                    if isinstance(first_element[0], int):
+                        prompt = pipeline.tokenizer.decode(first_element)
```

Comment:
how does decode work when decoding a single token at a time? it won't be 1:1 with the plaintext prompt right?

Reply:
this is decoding all the tokens at once - first_element is a `List[int]` after this conditional checks

Reply:
any consideration for special characters? I believe we have a specific param in the pipeline

Reply:
i think we do want to preserve special tokens, and `skip_special_tokens` is False by default